src/Host/Client/Impl/Session/RSession.cs

```diff
@@ -38,7 +37,7 @@ internal sealed class RSession : IRSession, IRCallbacks {
         public event EventHandler<EventArgs> Disconnected;
         public event EventHandler<EventArgs> Disposed;
         public event EventHandler<EventArgs> DirectoryChanged;
-        public event EventHandler<EventArgs> BeforePackagesInstalled;
+        public event AsyncEventHandler<EventArgs> BeforePackagesInstalledAsync;
```

Comment:
Option 1 is to leave async event alone. There is only one listener anyway. We are sorting out dependencies in master and my plan is to get rid of Support assembly anyway.